src/world/World.h

```diff
@@ -37,10 +37,20 @@ class World {
 public:
     static constexpr const int MAX_SEARCH_RESULTS = 10;
 
+    enum NodeVisitFilter {
+        VISIT_AIRPORTS =    0b1,
+        VISIT_AIRFIELDS =   0b01,
+        VISIT_FIXES =       0b001,
+        VISIT_NAVAIDS =     0b0001,
+        VISIT_USER_FIXES =  0b00001,
+        VISIT_EVERYTHING = (VISIT_AIRPORTS | VISIT_AIRFIELDS | VISIT_NAVAIDS | VISIT_FIXES | VISIT_USER_FIXES)
+    };
+
     using NodeAcceptor = std::function<void(const world::NavNode &node)>;
     using Connection = std::pair<std::shared_ptr<NavEdge>, std::shared_ptr<NavNode>>;
 
-    virtual void visitNodes(const world::Location &upLeft, const world::Location &lowRight, NodeAcceptor f) = 0;
+    virtual int countNodes(const world::Location &upLeft, const world::Location &lowRight) = 0;
+    virtual void visitNodes(const world::Location &upLeft, const world::Location &lowRight, NodeAcceptor calllback, NodeVisitFilter filter) = 0;
```

Comment:
Is the new filter used at any calling site yet? Because I don't think this kind of flag usage is legal in C++ - combining two enum values with an operator will result in an int, and in C++ there is no implicit cast back to an enum. Minimal example here: https://godbolt.org/z/75Ph58nK1

Reply:
Also, all enum values are the same: 0b1 == 0b0001 :-)

Reply:
No, the filter isn't used yet, only the combined value is passed in. And thanks for spotting the enum bitmask mistake. If I'd tested the filter I might have seen those issues first!
I'll change the enum to a set of const values, and fix the value definitions too.